terraform/projects/app-ecs-services/alertmanager-service.tf

```diff
@@ -0,0 +1,123 @@
+/**
+* ECS service that runs alertmanager
+*
+* This service consists of two containers.  The first, `s3-config-grabber`, fetches alertmanager configuration from our config S3 bucket, and stores it on a shared volume.  Then `alertmanager` runs and consumes that config.
+*
+* There is a known race condition between the two tasks - there is no guarantee that `s3-config-grabber` will grab the config before `alertmanager` starts.
+*
+*/
+
+## IAM roles & policies
+
+resource "aws_iam_role" "alertmanager_task_iam_role" {
+  name = "${var.stack_name}-alertmanager-task"
+  path = "/"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "alertmanager_policy_doc" {
+  statement {
+    sid = "GetAlertmanagerFiles"
+
+    resources = ["arn:aws:s3:::${aws_s3_bucket.config_bucket.id}/alertmanager/*"]
+
+    actions = [
+      "s3:Get*",
+    ]
+  }
+
+  statement {
+    sid = "ListConfigBucket"
+
+    resources = ["arn:aws:s3:::${aws_s3_bucket.config_bucket.id}"]
+
+    actions = [
+      "s3:List*",
+    ]
+  }
+}
+
+#I need to refactor to ensure comppatability with other domains=
+resource "aws_route53_record" "alertmanager_dns" {
```

Comment:
Can this be moved to the infra-networking-route53 project please.

Reply:
I am not sure. The only problem I have with that is that I wanted the domain name to  be close to the service. I understand that we have sub-zone created within main.tf of route53 folder. If we keep the DNS record setting separate from service file. It means that when deleting services a stale DNS record will remain unless we perform a destroy on the DNS records folder. 

I think there should be some static values within the DNS folder, this will be something like prom-1 & alert-1. We should use the ability to set a custom value closer to the service. There is also an issue that if you run apply-single on the service dir it will not update infra-route53 folder.

I am happy to move this into that folder but I need you consider this point first.

Reply:
I ended up moving this, I think it is better for the time being until a use case comes up.